src/Platform/Microsoft.Testing.Platform/OutputDevice/TerminalOutputDevice.cs

```diff
@@ -404,57 +408,73 @@ public async Task ConsumeAsync(IDataProducer dataProducer, IData value, Cancella
                              testNodeStateChanged.TestNode.DisplayName,
                              TestOutcome.Fail,
                              duration,
-                             errorMessage: failedState.Exception?.Message ?? failedState.Explanation,
-                             failedState.Exception?.StackTrace,
+                             errorMessage: string.Join(_environment.NewLine, exceptionMessages),
+                             errorStackTrace: string.Join(_environment.NewLine, exceptionStacktraces),
                              expected: failedState.Exception?.Data["assert.expected"] as string,
                              actual: failedState.Exception?.Data["assert.actual"] as string);
                         break;
 
                     case TimeoutTestNodeStateProperty timeoutState:
+                        exceptions = GetExceptionsInThrownOrder(timeoutState.Exception);
+                        exceptionMessages = GetExceptionMessages(exceptions, timeoutState.Explanation);
+                        exceptionStacktraces = GetExceptionStackTraces(exceptions);
+
                         _terminalTestReporter.TestCompleted(
                              _assemblyName,
                              _targetFramework,
                              _shortArchitecture,
                              testNodeStateChanged.TestNode.DisplayName,
                              TestOutcome.Timeout,
                              duration,
-                             errorMessage: timeoutState.Exception?.Message ?? timeoutState.Explanation,
-                             timeoutState.Exception?.StackTrace,
+                             errorMessage: string.Join(_environment.NewLine, exceptionMessages),
+                             errorStackTrace: string.Join(_environment.NewLine, exceptionStacktraces),
                              expected: null,
                              actual: null);
                         break;
 
                     case CancelledTestNodeStateProperty cancelledState:
+                        exceptions = GetExceptionsInThrownOrder(cancelledState.Exception);
+                        exceptionMessages = GetExceptionMessages(exceptions, cancelledState.Explanation);
+                        exceptionStacktraces = GetExceptionStackTraces(exceptions);
+
                         _terminalTestReporter.TestCompleted(
                              _assemblyName,
                              _targetFramework,
                              _shortArchitecture,
                              testNodeStateChanged.TestNode.DisplayName,
                              TestOutcome.Canceled,
                              duration,
-                             errorMessage: cancelledState.Exception?.Message ?? cancelledState.Explanation,
-                             cancelledState.Exception?.StackTrace,
+                             errorMessage: string.Join(_environment.NewLine, exceptionMessages),
```

Comment:
the receiving api can be changed to take array of strings, (we use it both here and behind a remote call in `dotnet test` so we don't want exception objects). 

That will allow us to render the erros and stack traces under each other, so you don't have to jump back and forth.

Reply:
Sounds good. Do you want to take over this pr because I can't actually build it locally so it makes Dev a bit harder

Reply:
Sure.